src/mfa/totp/service.ts

```diff
@@ -17,17 +17,19 @@ export function generateSecret(): string {
 }
 
 export async function save(totpDevice: NewTotpDevice): Promise<TotpDevice> {
-  const query = 'INSERT INTO user_totp SET ?, created = UNIX_TIMESTAMP()';
-
   const newTotpDeviceRecord: Partial<UserTotpDeviceRow> = {
     user_id: totpDevice.user.id,
     secret: totpDevice.secret,
   };
 
-  const result = await database.query(query, [newTotpDeviceRecord]);
+  const connection = await database.getConnection();
+
+  const result = await connection<UserTotpDeviceRow>('user_totp')
+    .insert(newTotpDeviceRecord, 'id')
+    .returning('id')
```

Comment:
I'm fairly certain `.returning()`  breaks in MySQL. I recently ran into this myself. Knex (for me) throws a warning that it's not supported in MySQL.

Reply:
Also this might be missing a value for the `created` field

Reply:
Yeah, so line 28's additional parameter is what allows MySQL to return as expected, so to support both databases you have to do line 28 and line 29, `.returning('id')` wont return the id in MySQL on its own.